sentry-android-core/src/main/java/io/sentry/android/core/AppComponentsBreadcrumbsIntegration.java

```diff
@@ -0,0 +1,95 @@
+package io.sentry.android.core;
+
+import android.content.ComponentCallbacks;
+import android.content.Context;
+import android.content.res.Configuration;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.sentry.android.core.util.DeviceOrientations;
+import io.sentry.core.Breadcrumb;
+import io.sentry.core.IHub;
+import io.sentry.core.Integration;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import io.sentry.core.protocol.Device;
+import io.sentry.core.util.Objects;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Locale;
+import org.jetbrains.annotations.NotNull;
+
+public final class AppComponentsBreadcrumbsIntegration
+    implements Integration, Closeable, ComponentCallbacks {
+
+  private final @NotNull Context context;
+  private @Nullable IHub hub;
+  private @Nullable SentryAndroidOptions options;
+
+  public AppComponentsBreadcrumbsIntegration(final @NotNull Context context) {
+    this.context = Objects.requireNonNull(context, "Context is required");
+  }
+
+  @Override
+  public void register(final @NotNull IHub hub, final @NotNull SentryOptions options) {
+    this.hub = Objects.requireNonNull(hub, "Hub is required");
+    this.options =
+        Objects.requireNonNull(
+            (options instanceof SentryAndroidOptions) ? (SentryAndroidOptions) options : null,
+            "SentryAndroidOptions is required");
+
+    this.options
+        .getLogger()
+        .log(
+            SentryLevel.DEBUG,
+            "AppComponentsBreadcrumbsIntegration enabled: %s",
+            this.options.isEnableAppComponentBreadcrumbs());
+
+    if (this.options.isEnableAppComponentBreadcrumbs()) {
+      context.registerComponentCallbacks(this);
+      options.getLogger().log(SentryLevel.DEBUG, "AppComponentsBreadcrumbsIntegration installed.");
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    context.unregisterComponentCallbacks(this);
+
+    if (options != null) {
+      options.getLogger().log(SentryLevel.DEBUG, "AppComponentsBreadcrumbsIntegration removed.");
+    }
+  }
+
+  @SuppressWarnings("deprecation")
+  @Override
+  public void onConfigurationChanged(@NonNull Configuration newConfig) {
+    if (hub != null) {
+      final Device.DeviceOrientation deviceOrientation =
+          DeviceOrientations.getOrientation(context.getResources().getConfiguration().orientation);
+
+      String orientation;
+      if (deviceOrientation != null) {
+        orientation = deviceOrientation.name().toLowerCase(Locale.ROOT);
+      } else {
+        orientation = "undefined";
+      }
+
+      final Breadcrumb breadcrumb = new Breadcrumb();
+      breadcrumb.setType("navigation");
+      breadcrumb.setCategory("app.deviceOrientation");
```

Comment:
What do you think if we use dot notation for everything?
```suggestion
      breadcrumb.setCategory("app.device.orientation");
```

Reply:
@HazAT `app.orientation` then, frontend only handles `category.subcategory`, what do you think? or even `device.orientation`

Reply:
I see, yeah then maybe just `device.orientation`